src/Symfony/Component/Validator/Constraints/TypeValidator.php

```diff
@@ -29,6 +29,9 @@ class TypeValidator extends ConstraintValidator
         'float' => 'is_float',
         'double' => 'is_float',
         'real' => 'is_float',
+        'number' => 'is_int || is_float',
```

Comment:
```suggestion
        'number' => 'is_int || is_float && !is_nan',
```

Reply:
for what it's worth, psalm calls it `numeric` and it's a super type of int or float and `numeric-string`

Reply:
numeric is already a thing for validator, we don't have to follow psalm (we don't already)

number != is-not-a-number

any other opinion on this?

Reply:
excusi, i missed existing `numeric` below

having both numeric + number is fine :+1: but i'd prefer consistent `finite-*` for all cases